### PR TITLE
feat: only save the last 5 rounds when reconnecting

### LINKED ISSUE #95: Store locally only the last X transaction logs
Performance seems to drop when transaction logs stack. Therefore, we should keep only the last X (the initialization,deployment and maybe the next 5 rounds)

### PR DESCRIPTION
**Pull request by martinkaintas**
closes #95 
With this PR we now only save logs for the last 5 rounds when reconnecting.

### CODE REVIEW

File: client/src/utils/game-channel/game-channel.ts [hunk 1]
```diff
@@ -54,6 +57,7 @@ export class GameChannel {
   timerStartTime = -1;
   lastOffChainTxTime = -1;
   channelIsClosing = false;
+  didReconnect = false;
```
Comment (Kalovelo) [inline client/src/utils/game-channel/game-channel.ts:60]:
```suggestion
```

We can use `const didReconnect = !!localStorage.getItem('gameState');` to check if reconnection will be triggered and avoid increasing the total object fields of the game state

File: client/src/components/end-screen/end-screen.vue [hunk 2]
```diff
@@ -3,14 +3,27 @@ import { computed } from 'vue';
...
+import { TransactionLog } from '../transaction/transaction.vue';
+import { useTransactionsStore } from '../../stores/transactions';
 
 const channel = useChannelStore().channel as GameChannel;
 
-const transactions = useTransactionsStore()
-  .userTransactions.flat()
-  .filter((tx) => tx.onChain === false);
+let offChainTxNo = 0;
+if (channel.didReconnect) {
```
Comment (Kalovelo) [inline client/src/components/end-screen/end-screen.vue:13]:
```suggestion
let offChainTxNo = 0;
const didReconnect = !!localStorage.getItem('gameState');
if (didReconnect) {
```
No need for it to be ref as we will only check it on first render

File: client/src/components/transaction-list/transaction-list.vue [hunk 3]
```diff
@@ -16,6 +18,8 @@ const terminalEl = ref<Element>();
 const transactionStore = useTransactionsStore();
 const { userTransactions, botTransactions } = storeToRefs(transactionStore);
 
+const channel = useChannelStore().channel;
+
```
Comment (Kalovelo) [inline client/src/components/transaction-list/transaction-list.vue:22]:
```suggestion
const channel = useChannelStore().channel;
const didReconnect = !!localStorage.getItem('gameState');
```

No need for it to be ref as we will only check it on first render

File: client/src/components/transaction-list/transaction-list.vue [hunk 4]
```diff
@@ -52,7 +56,15 @@ onUpdated(() => {
           <span> {{ isFullscreen ? 'Minimize' : 'Expand' }} </span>
         </button>
       </div>
-      <div class="transactions-list" v-if="botTransactions.length > 0">
+      <div
+        class="transactions-list"
+        v-if="Object.keys(botTransactions).length > 0"
+      >
+        <Notification
+          v-if="channel?.didReconnect"
```
Comment (Kalovelo) [inline client/src/components/transaction-list/transaction-list.vue:64]:
```suggestion
          v-if="didReconnect"
```

File: client/src/components/end-screen/end-screen.vue (hunk 2 above)
Comment (martinkaintas) [reply client/src/components/end-screen/end-screen.vue:13]: What ref?

**Pull request merged**